### PR TITLE
Add MH decorator & websites

### PR DESCRIPTION
**Pull request by MikeZeDev**

### CODE REVIEW

File: web/src/engine/websites/iqiyi.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+function MangaExtractor(anchor: HTMLAnchorElement) {
+    const id = anchor.pathname;
+    const title = anchor.nextElementSibling.textContent.trim();
+    return { id, title };
+}
+
+@Common.MangaCSS(/^{origin}\/manhua\/detail_\S+\.html$/, '.detail-tit h1')
+@Common.MangasMultiPageCSS('/manhua/category/%E5%85%A8%E9%83%A8_-1_-1_9_{page}', 'ul.cartoon-hot-ul li.cartoon-hot-list a.cartoon-cover', 1, 1, 0, MangaExtractor)
+@Common.PagesSinglePageCSS('ul.main-container li.main-item img', MH.PageLinkExtractor)
```
Comment (ronny1982) [inline web/src/engine/websites/iqiyi.ts:26]: This page looks not like `MH`, does the default `PageLinkExtractor` from `Common` not work?

File: web/src/engine/websites/iqiyi.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const manhuaId = new URL(manga.Identifier, this.URI).href.match(/_(([a-z]|[0-9])*)/)[1];
+        const requestChaps = new FetchRequest(new URL(`/manhua/catalog/${manhuaId}/`, this.URI).href);
+        const { data } = await FetchJSON<APIChapters>(requestChaps);
+        return data.episodes.map(element => new Chapter(this, manga, `/manhua/reader/${manhuaId}_${element.episodeId}.html`, [element.episodeOrder, element.episodeTitle.trim()].join(" ")));
```
Comment (ronny1982) [inline web/src/engine/websites/iqiyi.ts:43]: Avoid double quotes, use single quotes instead

File: web/src/engine/websites/iqiyi.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+}
+function MangaExtractor(anchor: HTMLAnchorElement) {
+    const id = anchor.pathname;
+    const title = anchor.nextElementSibling.textContent.trim();
+    return { id, title };
+}
+
+@Common.MangaCSS(/^{origin}\/manhua\/detail_\S+\.html$/, '.detail-tit h1')
```
Comment (ronny1982) [inline web/src/engine/websites/iqiyi.ts:24]: The `\S+` term allows non-alphanumeric characters such as `/` or `-` is this by intention?

**Review by ronny1982 — APPROVED** [review]

File: web/src/engine/websites/iqiyi.ts (hunk 3 above)
Comment (MikeZeDev) [reply web/src/engine/websites/iqiyi.ts:24]: Ah missed this one.

File: web/src/engine/websites/iqiyi.ts (hunk 1 above)
Comment (MikeZeDev) [reply web/src/engine/websites/iqiyi.ts:26]: 1) iqiyi is extending MH in Hakuneko so it ended up in that PR, i think there is a reason for that.
2) No it wouldnt work

image

**Pull request merged**